app/routes/shared/__test__/pdfDownloadLoader.test.ts

```diff
@@ -0,0 +1,96 @@
+import type { LoaderFunctionArgs } from "@remix-run/node";
+import { PDFDocument } from "pdf-lib";
+import { pdfDownloadLoader } from "../pdfDownloadLoader";
+
+// Convert Buffers to Uint8Array for compatibility with Vitest
+// Vitest expects Uint8Array inputs for certain methods in pdf-lib.
+function wrapPdfDocumentLoad() {
```

Comment:
nit: would rename these functions something like 
```suggestion
function spyOnPdfDocumentLoad() {
```
as it wasn't immediately clear what these were doing

Reply:
Example: `pdfKitBuffer` in `app/domains/beratungshilfe/services/pdf/index.ts` is of type `Buffer`. Vitest expects is to be `Uint8Array` (please don't ask me why) which is why it needs to be wrapped like this `new Uint8Array(pdfBuffer)` before passing it into `PDFDocument.load`. Instead of touching the code in several places just for this one test, I've decided to wrap on demand inside of the test. So I think the name is quite appropriate. wdyt?